searchlib/src/vespa/searchlib/attribute/attributevector.cpp

```diff
@@ -769,6 +770,31 @@ AttributeVector::logEnumStoreEvent(const char *reason, const char *stage)
     EV_STATE(eventName.c_str(), jstr.toString().data());
 }
 
+void
+AttributeVector::drain_hold(uint64_t hold_limit)
+{
+    incGeneration();
+    for (int retry = 0; ; ++retry) {
+        removeAllOldGenerations();
+        updateStat(true);
+        if (_status.getOnHold() <= hold_limit) {
+            return;
+        }
+        std::this_thread::sleep_for(retry < 20 ? 20ms : 100ms);
+    }
+}
+
+void
+AttributeVector::update_config(const Config& cfg)
```

Comment:
We should add a unit test for this on the attribute vector level.

Reply:
It is unit tested, c.f. attribute_compaction_test.cpp.